### PR TITLE
Add constructor taking tenant id as parameter for AzureCredentials constructor when using MSI

### LINKED ISSUE #661: [BUG]Cannot query Azure AD with system managed identity
**Describe the bug**
Authenticating with `new AzureCredentials(new MSILoginInformation(MSIResourceType.AppService), AzureEnvironment.AzureGlobalCloud);` causes the tenant id to be null, which causes subsequent requests to Azure AD to fail.

***Exception or Stack Trace***
```
"Microsoft.Rest.ValidationException: 'this.Client.TenantID' cannot be null.
   at Microsoft.Azure.Management.Graph.RBAC.Fluent.ServicePrincipalsOperations.ListWithHttpMessagesAsync(ODataQuery`1 odataQuery, Dictionary`2 customHeaders, CancellationToken cancellationToken)
   at Microsoft.Azure.Management.Graph.RBAC.Fluent.ServicePrincipalsOperationsExtensions.ListAsync(IServicePrincipalsOperations operations, ODataQuery`1 odataQuery, CancellationToken cancellationToken)
   at Microsoft.Azure.Management.Graph.RBAC.Fluent.ServicePrincipalsImpl.<ListAsync>b__10_0(CancellationToken cancellation)
   at Microsoft.Azure.Management.ResourceManager.Fluent.Core.PagedCollection`2.LoadPageWithWrapModelAsync(Func`2 listInnerAsync, Func`3 listInnerNext, Func`3 wrapModelAsync, Boolean loadAllPages, CancellationToken cancellationToken)
   at Microsoft.Azure.Management.Graph.RBAC.Fluent.ServicePrincipalsImpl.ListAsync(Boolean loadAllPages, CancellationToken cancellationToken)
   at FunctionApp4.Function1.Run(Ht
tpRequest req, ILogger log) in C:\\Users\\Ged\\source\\repos\\FunctionApp4\\FunctionApp4\\Function1.cs:line 26
   at Microsoft.Azure.WebJobs.Host.Executors.FunctionInvoker`2.InvokeAsync(Object instance, Object[] arguments) in C:\\projects\\azure-webjobs-sdk-rqm4t\\src\\Microsoft.Azure.WebJobs.Host\\Executors\\FunctionInvoker.cs:line 52"
```

**To Reproduce**

1. Create a new Azure Function
2. Enable system managed identity for the app service, hosting the function
3. Paste the code snippet into the function
4. Execute the function

***Code Snippet***
```
AzureCredentials credentials = new AzureCredentials(new MSILoginInformation(MSIResourceType.AppService), AzureEnvironment.AzureGlobalCloud);
Azure.IAuthenticated azure = Azure.Authenticate(credentials);
IAzure authenticated = await azure.WithDefaultSubscriptionAsync();
IPagedCollection<IServicePrincipal> spn = await authenticated.AccessManagement.ServicePrincipals.ListAsync();
```

**Expected behavior**
`IAzure.AccessManagement.ServicePrincipals.ListAsync()` should return a list of Service Principals, in the tenant where the managed identity exists, when authenticating with a system managed identity.

**Screenshots**
N/A

**Setup (please complete the following information):**
 - OS: Windows 7
 - IDE : Microsoft Visual Studio Community 2019
 - Version of the Library used: 1.22.0

**Additional context**
N/A

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [X] Bug Description Added
- [X] Repro Steps Added
- [X] Setup information Added

### PR DESCRIPTION
**Pull request by michaelelleby**
Fixes #661

### CODE REVIEW

File: src/ResourceManagement/ResourceManager/Authentication/AzureCredentials.cs [hunk 1]
```diff
@@ -81,6 +81,12 @@ public AzureCredentials(MSILoginInformation msiLoginInformation, AzureEnvironmen
             this.msiTokenProviderFactory = new MSITokenProviderFactory(msiLoginInformation);
         }
 
+        public AzureCredentials(MSILoginInformation msiLoginInformation, string tenantId, AzureEnvironment environment)
```
[the remark this answers is not in the document]
Comment (yaohaizh) [reply src/ResourceManagement/ResourceManager/Authentication/AzureCredentials.cs]: Make the teantId as optional parameter as the last one and default value to null would be better, but not provide another API. Thanks.

**Comment by yaohaizh** [comment]
@michaelelleby could you help update this PR? Thanks.

**Comment by michaelelleby** [comment]
@yaohaizh I have updated the PR, to avoid introducing any breaking changes to the SDK.

**Comment by yaohaizh** [comment]
Thanks for your contributions!

**Pull request merged**